### PR TITLE
 Implement a test for chunking ng fail due to a lock

### PR DESCRIPTION
**Pull request by TheOneRing**
The unit test is supposed to fail

### CODE REVIEW

**Comment by fmoc** [comment]
Please rebase.

Also, your test isn't running on the CI.

**Comment by TheOneRing** [comment]
> Please rebase on `master`.
> 
> Also, your test isn't working in the CI.

Please read the description 🤣

File: test/testchunkingng.cpp [hunk 1]
```diff
@@ -628,7 +628,45 @@ private slots:
         QCOMPARE(fakeFolder.currentRemoteState().find("A/a0")->contentSize, size + 1);
     }
 
+    // the final move fails but the succeeding sync recovers and finishes the upload
+    void testFinalMoveLocked()
+    {
+        FakeFolder fakeFolder { FileInfo::A12_B12_C12_S12() };
+        const auto size = 15_mb;
```
Comment (fmoc) [inline test/testchunkingng.cpp:635]: This unit looks like "millibit"...

File: test/testchunkingng.cpp [hunk 2]
```diff
@@ -628,7 +628,45 @@ private slots:
...
+         * Similar to the Folder class we will trigger a new sync but without the local discovery.
+         * The test ensures that we properly continue the download and only perform the final move.
+         */
+        fakeFolder.localModifier().insert(QStringLiteral("A/a0"), size);
+        QVERIFY(fakeFolder.applyLocalModificationsAndSync());
+        QVERIFY(fakeFolder.currentLocalState() != fakeFolder.currentRemoteState());
+        QCOMPARE(counter.nMOVE, 1);
+
+        counter.reset();
+        fakeFolder.setServerOverride(counter.functor());
```
Comment (fmoc) [inline test/testchunkingng.cpp:660]: Does `setServerOverride` have any non-obvious side effects?

**Review by fmoc — APPROVED** [review]

File: test/testchunkingng.cpp (hunk 1 above)
Comment (TheOneRing) [reply test/testchunkingng.cpp:635]: patches welcome

File: test/testchunkingng.cpp (hunk 2 above)
Comment (TheOneRing) [reply test/testchunkingng.cpp:660]: No idea

**Pull request merged**